components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -923,6 +923,7 @@ public long contentLength() throws IOException {
                                 @Override
                                 public void writeTo(@Nonnull BufferedSink sink) throws IOException {
                                     sink.writeAll(Okio.source(requestInfo.content));
+                                    requestInfo.content.reset(); // unsafe
```

Comment:
unsafe operation. Adding for discussion purposes on alternatives allowing multiple reads from the stream without copying contents into an in-memory buffer

Reply:
Added condition to check if resettable, before performing reset